src/main/java/randoop/generation/Impurity.java

```diff
@@ -0,0 +1,518 @@
+      boolean explicitCast) {
+    CallableOperation callableOperation = createCallableOperation(executable, explicitCast);
+    NonParameterizedType declaringType = new NonParameterizedType(executable.getDeclaringClass());
+    List<Type> inputTypeList = getInputTypeList(executable, declaringType);
+    TypeTuple inputType = new TypeTuple(inputTypeList);
+    TypedOperation typedOperation =
+        new TypedClassOperation(callableOperation, declaringType, inputType, outputType);
+    List<Integer> inputIndex = calculateInputIndex(sequence, inputTypeList);
+    fuzzStatementOffset.increment(inputTypeList.size());
+    List<Sequence> sequenceList = Collections.singletonList(sequence);
+    return Sequence.createSequence(typedOperation, sequenceList, inputIndex);
+  }
+
+  /**
+   * Create a callable operation for fuzzing an object of a given type using the given method.
+   *
+   * @param executable the method to be invoked to fuzz the object
+   * @param explicitCast whether to perform an explicit cast for the right-hand side of the fuzzing
+   *     statement
+   * @return a callable operation for fuzzing an object of a given type using the given method
+   */
+  private static CallableOperation createCallableOperation(
+      Executable executable, boolean explicitCast) {
+    if (executable instanceof Method) {
+      return new MethodCall((Method) executable, explicitCast);
+    } else {
+      return new ConstructorCall((Constructor<?>) executable);
+    }
+  }
+
+  /**
+   * Determine the output type of the given method.
+   *
+   * @param executable the method to determine the output type of
+   * @return the output type of the given method
+   */
+  private static Type determineOutputType(Executable executable) {
+    Class<?> outputClass;
+    if (executable instanceof Method) {
+      outputClass = ((Method) executable).getReturnType();
+    } else {
+      outputClass = ((Constructor<?>) executable).getDeclaringClass();
+    }
+    return outputClass.isPrimitive()
+        ? PrimitiveType.forClass(outputClass)
+        : new NonParameterizedType(outputClass);
+  }
+
+  /**
+   * Get the list of input types for the given method.
+   *
+   * @param executable the method to get the input types of
+   * @param declaringType the type that declares the given method
+   * @return the list of input types for the given method
+   */
+  private static List<Type> getInputTypeList(
+      Executable executable, NonParameterizedType declaringType) {
+    List<Type> inputTypeList = new ArrayList<>();
+    if (!Modifier.isStatic(executable.getModifiers()) && executable instanceof Method) {
+      inputTypeList.add(declaringType);
+    }
+    for (Class<?> clazz : executable.getParameterTypes()) {
+      inputTypeList.add(
+          clazz.isPrimitive() ? PrimitiveType.forClass(clazz) : new NonParameterizedType(clazz));
+    }
+    return inputTypeList;
+  }
+
+  /**
+   * Calculate the index of the input parameters in the given sequence.
```

Comment:
This makes the assumption that "the input parameters" are the last _n_ statements in the sequence.  When there is an important precondition like that, please state it in the documentation.  Also, what does "the input parameters" mean?